pkg/frontend/transport/handler.go

```diff
@@ -252,6 +265,97 @@ func (f *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func getQueryDetail(url string, values url.Values) QueryDetail {
+	urlValues := parseURLValues(values)
+	t := InstantQuery
+	if strings.HasSuffix(url, "query_range") {
+		t = RangeQuery
+	}
+	return QueryDetail{
+		QueryString: urlValues.query,
+		StepInt:     urlValues.stepInt,
+		StartInt:    urlValues.startInt,
+		EndInt:      urlValues.endInt,
+		TsInt:       urlValues.tsInt,
+		QueryType:   t,
+	}
+}
+
+type urlValues struct {
+	query                            string
+	stepInt, startInt, endInt, tsInt int64
+}
+
+func parseURLValues(values url.Values) urlValues {
+	query := values.Get("query")
+	step := values.Get("step")
+	start := values.Get("start")
+	end := values.Get("end")
+	ts := values.Get("time")
+	var (
+		tsInt, startInt, endInt, stepInt int64
+		err                              error
+	)
+	if len(ts) > 0 {
```

Comment:
A better way here is, we can check `len(step) > 0` to see if it is a range query. Then for instant query, we just need to parse ts and can omit the rest 3.
Also for `time` param, we might need to set a default value. https://github.com/prometheus/prometheus/blob/main/web/api/v1/api.go#LL401C13-L401C27

Reply:
Changed that.